lldb/test/API/lang/swift/archetype_in_expression/TestArchetypeInExpression.py

```diff
@@ -0,0 +1,37 @@
+import lldb
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test.decorators import *
+import lldbsuite.test.lldbutil as lldbutil
+import unittest2
+
+
+class TestBoundOpaqueArchetype(TestBase):
+
+    @swiftTest
+    def test(self):
+        """Tests that a user can refer to the archetypes in their expressions"""         
+        self.build()
+
+        # Check that you can refer to all types and function's archetypes.
+        _, process, _, breakpoint = lldbutil.run_to_source_breakpoint(self, 
+                "break here", lldb.SBFileSpec("main.swift"))
+        self.expect("e First.self", substrs=["Int.Type"])
+        self.expect("e Second.self", substrs=["Double.Type"])
+        self.expect("e Third.self", substrs=["Bool.Type"])
+        self.expect("e T.self", substrs=["String.Type"])
+        self.expect("e U.self", substrs=["[Int].Type"])
+
+        # Check that referring to a shadowed archetype works correctly.
+        lldbutil.continue_to_breakpoint(process, breakpoint)
+        self.expect("e T.self", substrs=["String.Type"])
+
+        # Check that you refer to archetypes in nested generic functions.
+        lldbutil.continue_to_breakpoint(process, breakpoint)
+        self.expect("e T.self", substrs=["Bool.Type"])
+        self.expect("e U.self", substrs=["Double.Type"])
+
+        # Check that you refer to archetypes in nested generic functions with shadowed archetypes.
+        lldbutil.continue_to_breakpoint(process, breakpoint)
+        self.expect("e T.self", substrs=["String.Type"])
+        lldbutil.continue_to_breakpoint(process, breakpoint)
+        self.expect("e T.self", substrs=["Int.Type"])
```

Comment:
Should we explicitly test setting a conditional breakpoint on a `T.self == Int.Type`?